contrib/passwd.bzl

```diff
@@ -12,24 +12,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-def _impl(ctx):
-  """Core implementation of passwd_file."""
+PasswdFileContentProvider = provider()
```

Comment:
You can define the fields in the provider using fields notation thereby avoiding runtime error.
https://docs.bazel.build/versions/master/skylark/lib/globals.html#provider